src/function/scalar/sequence/currval.cpp

```diff
@@ -0,0 +1,94 @@
+#include "duckdb/function/scalar/sequence_functions.hpp"
+
+#include "duckdb/catalog/catalog.hpp"
+#include "duckdb/catalog/catalog_entry/sequence_catalog_entry.hpp"
+#include "duckdb/common/exception.hpp"
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "duckdb/execution/expression_executor.hpp"
+#include "duckdb/catalog/catalog.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/transaction/transaction.hpp"
+#include "duckdb/common/vector_operations/unary_executor.hpp"
+
+namespace duckdb {
+
+struct CurrvalBindData : public FunctionData {
+	//! The client context for the function call
+	ClientContext &context;
+	SequenceCatalogEntry *sequence;
+
+	CurrvalBindData(ClientContext &context, SequenceCatalogEntry *sequence) : context(context), sequence(sequence) {
+	}
+
+	unique_ptr<FunctionData> Copy() override {
+		return make_unique<CurrvalBindData>(context, sequence);
+	}
+};
+
+static int64_t CurrSequenceValue(Transaction &transaction, SequenceCatalogEntry *seq) {
+	lock_guard<mutex> seqlock(seq->lock);
+	int64_t result;
+	if (seq->usage_count == 0u) {
+		throw SequenceException("currval: sequence has not been used yet.");
```

Comment:
Can we copy the Postgres error message here exactly (e.g. `currval of sequence is not yet defined in this session`).

Reply:
the error messages for nextval start with "nextval: ..." so how about "currval: currval of sequence is not yet defined in this session" or "currval: sequence is not yet defined in this session"

Reply:
Sure, sounds good.